files/image_config/monit/container_checker

```diff
@@ -0,0 +1,99 @@
+    except OSError as os_err:
+        print("Failed to execute the command '{}'. Error: '{}'".format(command, os_err))
+        sys.exit(2)
+    except ValueError as val_err:
+        print("Failed to execute the command '{}'. Error: '{}'".format(command, val_err))
+        sys.exit(3)
+
+    return command_stdout.split("\n")
+
+
+def get_expected_running_containers():
+    """
+    @summary: This function will get the expected running containers by following the rule:
+              The 'state' field of container in 'FEATURE' table should not be 'disabled'. Then
+              if the device has Multi-ASIC, this function will get container list by determining the
+              value of field 'has_global_scope', the number of ASICs and the value of field
+              'has_per_asic_scope'. If the device has single ASIC, the container name was put into
+              the list.
+    @return:  A set which contains the expected running containers.
+    """
+    config_db = swsssdk.ConfigDBConnector()
+    config_db.connect()
+    feature_table = config_db.get_table("FEATURE")
+
+    expected_running_containers = set()
+
+    for container_name in feature_table.keys():
+        if feature_table[container_name]["state"] != "disabled":
+            if multi_asic.is_multi_asic():
+                if feature_table[container_name]["has_global_scope"] == "True":
+                    expected_running_containers.add(container_name)
+                if feature_table[container_name]["has_per_asic_scope"] == "True":
+                    num_asics = multi_asic.get_num_asics()
+                    for asic_id in range(num_asics):
+                        expected_running_containers.add(container_name + str(asic_id))
+            else:
+                expected_running_containers.add(container_name)
+
+    return expected_running_containers
+
+
+def get_current_running_containers():
+    """
+    @summary: This function will get the current running container list by analyzing the
+              output of command `docker ps`.
+    @return:  A set which contains the current running contianers.
+    """
+    running_containers = set()
+
+    command = "docker ps"
+    command_stdout = get_command_result(command)
+    for line in command_stdout:
+        if line and "CONTAINER ID" not in line:
+            running_containers.add(line.split()[-1].strip())
+
+    return running_containers
+
+
+def main():
+    """
+    @summary: This function will compare the difference between the current running containers
+              and the containers which were expected to run. If containers which were exepcted
+              to run were not running, then an alerting message will be written into syslog.
+    """
+    expected_running_containers = get_expected_running_containers()
+    current_running_containers = get_current_running_containers()
+
+    not_running_containers = expected_running_containers.difference(current_running_containers)
+    if not_running_containers:
```

Comment:
can we enhanced to this to see if we are not running extra containers. For eg:  in multi-asic we expect bgp0 and bgp1 but if docker ps O/P return bgp0, bgp1, bgp2. Can we catch this scenario also ?

Reply:
You mean that initially the expected running containers for example BGP container were BGP0 and BGP1, However, BGP0, BGP1, BGP2 and BGP3 are actually running. We should write an syslog message to show that extra containers BGP2 and BGP3 are running, right?

Reply:
yes